jerry-core/include/jerryscript-port.h

```diff
@@ -20,6 +20,32 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#if defined(_MSC_VER)
+#include <malloc.h>
+#endif
+
+#if defined(_MSC_VER)
+#define jerry_alloca(name, type, size) type *name = (type*)(alloca(sizeof(type)*size))
```

Comment:
As noticed already, cppcheck is picky about `alloca`. And by default cppcheck checks all preprocessor configurations, so even the MSVC-specific ones. It can either be tweaked not to check those configurations (perhaps by passing `-U_MSC_VER` to cppcheck, see tools/check-cppcheck.sh) or by adding `obsoleteFunctionsalloca` to the list of suppressed errors (see tools/cppcheck/suppressions-list).